curvefs/src/tools/curvefs_tool.h

```diff
@@ -90,42 +100,91 @@ template <class ChannelT, class ControllerT, class RequestT, class ResponseT,
           class ServiceT>
 class CurvefsToolRpc : public CurvefsTool {
  public:
-    CurvefsToolRpc(const std::string& command, const std::string& programe)
-        : CurvefsTool(command, programe) {}
+    CurvefsToolRpc(const std::string& command,
+                   const std::string& programe = std::string(kProgrameName),
+                   bool show = true)
+        : CurvefsTool(command, programe, show) {}
 
     int Init(const std::shared_ptr<ChannelT>& channel,
              const std::shared_ptr<ControllerT>& controller,
-             const std::shared_ptr<RequestT>& request,
+             const std::queue<RequestT>& requestQueue,
              const std::shared_ptr<ResponseT>& response,
-             const std::shared_ptr<ServiceT>& service_stub) {
+             const std::shared_ptr<ServiceT>& service_stub,
+             std::function<void(ControllerT*, RequestT*, ResponseT*)>
+                 service_stub_func) {
         channel_ = channel;
         controller_ = controller;
-        request_ = request;
+        requestQueue_ = requestQueue;
         response_ = response;
         service_stub_ = service_stub;
+        service_stub_func_ = service_stub_func;
+        InitHostsAddr();
         return 0;
     }
 
+    virtual int Init() {
+        channel_ = std::make_shared<ChannelT>();
+        controller_ = std::make_shared<ControllerT>();
+        // request[0] could be re-assigned by subclass
+        requestQueue_.push(RequestT());
```

Comment:
push an empty request, and set request fields by requestQueue_.front() is not straightforward